### ISSUE TITLE
grafana getting error "internal: expanding series: consistency check failed because some blocks were not queried" from cortex, store gateway pvc is full

### ISSUE THREAD

**Issue by danfinn** [cortexproject/cortex-helm-chart#497]
this appears to be the same issue as https://github.com/cortexproject/cortex/issues/4431 which was closed with no resolution or suggested workaround so I'm posting it again because I'm now hitting it.

I'm getting the following from grafana when querying cortex:

```
internal: expanding series: consistency check failed because some blocks were not queried: 01HVPPWCVQH6Q07DY0Y6YZW0P0 01HVPPX4VVJ1QGA80NYMZA5S0B 01HVPPX5BKY9AT9JAX1A7B0TWZ 01HVP94YBP2PKMG2SV0CYTFK03 01HVPG0NKPBQA1ZYKDV1YNYQVV 01HVPG1E3MJ7KKE6CW32AAFA3K 01HVPG25BKBGEXGP63M5JCDY6H 01HVPG1DKTH97ARYMMBNHZ6YWH 01HVP95PBVT2FWDS30CWKHQFYB 01HVP95PVKCFKQ95TSHZ73X5XA 01HVP96E3J44QR0YS1FC76702K 
```

and I see the following in my querier pod logs:

```
ts=2024-04-17T19:52:54.440956238Z caller=loader.go:142 level=info msg="loaded bucket index" user=dev2-cust-eastus duration=8.707502ms
ts=2024-04-17T19:52:54.442139376Z caller=spanlogger.go:87 org_id=dev2-cust-eastus method=blocksStoreQuerier.selectSorted level=warn msg="unable to get store-gateway clients while retrying to fetch missing blocks" err="no store-gateway instance left after filtering out excluded instances for block 01HVPG25BKBGEXGP63M5JCDY6H"
ts=2024-04-17T19:52:54.442177178Z caller=spanlogger.go:87 org_id=dev2-cust-eastus method=blocksStoreQuerier.selectSorted level=warn org_id=dev2-cust-eastus msg="failed consistency check" err=null
ts=2024-04-17T19:52:54.667815681Z caller=logging.go:86 level=warn traceID=733d4e4e33c2a418 orgID=dev2-cust-eastus msg="GET /api/prom/api/v1/query_range?end=1713383560&query=kube_pod_completion_time&start=1713361960&stats=&step=20 (500) 291.433678ms Response: \"\\x1f\\x8b\\b\\x00\\x00\\x00\\x00\\x00\\x00\\xff<\\xceϏ\\xa3 \\x14\\xc0\\xf1\\x7f\\x85x\\xde\\x03\\x8a\\xe2\\xea\\r\\xfc\\x19\\x89\\x06\\xaaA\\xf1f-\\xbbk\\xdaՎ\\xd8\\xcc4\\x93\\xf9\\xdf'M\\xed\\x9cH\\xf8\\xbc\\x97\\xf7\\xfd\\xb4\\xcc6l7c\\x85\\x96^\\xd7e\\xb5~=\\xdf\\xe6~\\xd5VhM\\xf3\\xa6\\xd7y\\xb8\\xbc\\xbe\\x1fc\\x1f\\xd7a>M\\xf3_`\\xf4:i\\x13\\x82q\\x99\\xcdd6=\\x8fw0\\xfe\\xd3\\xe3\\x19\\xfc\\x19\\xa6\\x8b>\\x81\\xa3\\x1e\\x87\\x9b\\xd1\\xc0,\\xff58^\\x96\\xf1l\\xc0\\xbb^5\\x98\\x97\\r\\xbc\\xdd\\x1e\\xeb\\xa7\\x10@;\\x97<s<\\xcah\\x96t\\x19Ǩ\\xf4\\x8a(V8\\xdfɎY\\x93\\a>9\\xa8\\xb2\\xa4U\\xdec\\xd5\\xee\\xc4\\xdbH\\x8a\\x1c\\v\\xe8\\xc7\\n*\\xac\\xfa\\x16r\\xb8S\\xe7Q\\xa6\\x02\\xd2\\x04\\x05\\xe9l\\xe2Sش\\xfd\\x93\\x02\\x8fK\\x16\\xa5L\\x04^S罏:\\xaf#;\\xb9\\x8ar\\x87\\xb32sj\\t#դ\\f\\xa2=\\x03V\\x8cSA\\xec^\\xb1XڪRB\\xcaWa\\x82\\xca\\xc2g,\\xc1Q\\x8b\\x1cBR\\x82\\xd8\\xcf-*\\x1b'm\\xe3\\x1a\\xc1\\xa8e\\xb9H\\x15\\xdd\\t'\\xa8p]q\\x80\\xaa\\xb6\\xd3\\xc8\\xc7>t\\xd8+ޕ\\xb2\\xb0EF~\\xc3J\\x95=\\xf1jH\\xad\\xafo\\x00\\x00\\x00\\xff\\xff\\x01\\x00\\x00\\xff\\xff*\\x83\\x9f3\\xad\\x01\\x00\\x00\" ws: false; Accept-Encoding: gzip; X-Scope-Orgid: dev2-cust-eastus; uber-trace-id: 733d4e4e33c2a418:3faedde89dc6c797:494197900fc17e13:0;
```

store gateway pod pvc is full and it's generating the following errors:
```
ts=2024-04-17T19:57:36.392646166Z caller=bucket.go:744 level=warn org_id=dev2-obsv-eastus msg="loading block failed" elapsed=3.967921ms id=01HVEHZF722VV36YVEATM4W95P err="create index header reader: write index header: new index reader: get TOC from object storage of 01HVEHZF722VV36YVEATM4W95P/index: create temp dir: mkdir /data/tsdb/dev2-obsv-eastus/01HVEHZF722VV36YVEATM4W95P: no space left on device"
```

The default size for the PVC for the store gateway pod looks to be 2 gig.  I've grown it to 10G and it looks like it has gotten cortex to recover however based on the other bug this will just continue to grow.

**Comment by nschad** [comment]
The default size of store-gateway is not representative of the size actually needed. We have 150Gi volumes.

Is that what you are asking?

**Comment by danfinn** [comment]
Yeah, I suppose this could be related to https://github.com/cortexproject/cortex/issues/5294 in that the default values do not seem to be suitable to production installs and the documentation seems to be lacking.

How should I size this PVC and how many replicas are suggested?  It also seems concerning that the other bug I linked to people are saying this PVC had grown to 2T and continue to grow.

For now with the help of the slack channel, I've increased the size of my PVCs for both store gateway and compactor and I've implemented a retention policy as well.
